lib/motion-spec/core.rb

```diff
@@ -26,67 +26,75 @@ module MotionSpec
     'colorized' => ColorizedOutput
   }
 
-  def self.add_context(context)
-    (@contexts ||= []) << context
-  end
-
-  def self.current_context_index
-    @current_context_index ||= 0
-  end
+  class << self
+    def add_context(context)
+      (@contexts ||= []) << context
+    end
 
-  def self.current_context
-    @contexts[current_context_index]
-  end
+    def current_context_index
+      @current_context_index ||= 0
+    end
 
-  def self.run(arg = nil)
-    unless respond_to?(:handle_specification_begin)
-      extend(Outputs[ENV['output']] || DEFAULT_OUTPUT_MODULE)
+    def current_context
+      @contexts[current_context_index]
     end
 
-    @timer ||= Time.now
+    def run(arg = nil)
```

Comment:
Unused method argument - `arg`. If it's necessary, use `_` or `_arg` as an argument name to indicate that it won't be used. You can also write as `run(*)` if you want the method to accept any arguments but don't care about them.